packages/next/build/webpack-config.ts

```diff
@@ -177,6 +177,100 @@ export default async function getBaseWebpackConfig(
 
   const devtool = dev ? 'cheap-module-source-map' : false
 
+  // Contains various versions of the Webpack SplitChunksPlugin used in different build types
+  const splitChunksConfigs: {
+    [propName: string]: webpack.Options.SplitChunksOptions
+  } = {
+    dev: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+      },
+    },
+    selective: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prod: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        commons: {
+          name: 'commons',
+          chunks: 'all',
+          minChunks: totalPages > 2 ? totalPages * 0.5 : 2,
+        },
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prodGranular: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        framework: {
+          name: 'framework',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+          priority: 40,
+        },
+        lib: {
+          test(module: { size: Function }): boolean {
+            return module.size() > 160000
+          },
+          name(module: { identifier: Function }): string {
+            let nodeModuleNameResults = /node_modules\/(.*)/.exec(
```

Comment:
it's super unlikely, but to avoid this catching `my_node_modules/foo`, I'd suggest:

```suggestion
            let nodeModuleNameResults = /(?:^|\/)node_modules\/(.*)/.exec(
```